test/battle/ability/commander.c

```diff
@@ -0,0 +1,420 @@
+#include "global.h"
+#include "test/battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(gMovesInfo[MOVE_ORDER_UP].additionalEffects[0].moveEffect == MOVE_EFFECT_ORDER_UP);
+}
+
+DOUBLE_BATTLE_TEST("(Commander) Commander will activate once Dondozo switches in")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_TATSUGIRI) { Ability(ABILITY_COMMANDER); }
+        PLAYER(SPECIES_DONDOZO);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { SWITCH(playerLeft, 2); }
+    } SCENE {
+        ABILITY_POPUP(playerRight, ABILITY_COMMANDER);
+        MESSAGE("Tatsugiri was swallowed by Dondozo and became Dondozo's commander!");
+    }
+}
+
+DOUBLE_BATTLE_TEST("(Commander) Commander increases all stats by 2 stages once it is triggered")
+{
+    GIVEN {
+        PLAYER(SPECIES_TATSUGIRI) { Ability(ABILITY_COMMANDER); }
+        PLAYER(SPECIES_DONDOZO);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN {  }
+    } SCENE {
+        ABILITY_POPUP(playerLeft, ABILITY_COMMANDER);
+        MESSAGE("Tatsugiri was swallowed by Dondozo and became Dondozo's commander!");
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, playerRight);
+        MESSAGE("Dondozo's Attack sharply rose!");
+        MESSAGE("Dondozo's Defense sharply rose!");
+        MESSAGE("Dondozo's Sp. Atk sharply rose!");
+        MESSAGE("Dondozo's Sp. Def sharply rose!");
+        MESSAGE("Dondozo's Speed sharply rose!");
+    }
+}
+
+DOUBLE_BATTLE_TEST("(Commander) When Commander is active moves targeted at Tatsugiri will fail")
```

Comment:
```suggestion
DOUBLE_BATTLE_TEST("Commander Tatsugiri avoids moves targetted towards it")
```

Reply:
Shouldn't all the Commander tests start with "Commander"?
So that you can run tests with TESTS="Commander" to run them all at once?

Reply:
That was my idea, ye. I modeled it after the the tera and dmax tests

Reply:
> Shouldn't all the Commander tests start with "Commander"? So that you can run tests with TESTS="Commander" to run them all at once?

Yeah, and that's where the name changes are doing, so we don't need the parenthesis.